File: src/backend/shoppingCartService/domain/dtos/CartDetailsDto.cs
namespace domain.dtos;

public record CartDetailsDto
{
    public Guid CartDetailsId { get; set; }
    public Guid CartHeaderId { get; set; }
    public CartHeaderDto CartHeader { get; init; }
    public string ProductId { get; init; }
    public ProductDto Product { get; set; }
    public int Count { get; set; }
}

File: src/backend/orderService/src/dtos/cartDetails.dto.ts
import { CartHeaderDto } from "./cartHeader.dto";
import { ProductDto } from "./product.dto";

export interface CartDetailsDto {
  cartDetailsId: string;
  cartHeaderId: string;
  cartHeaderDto: CartHeaderDto;
  productId: string;
  productDto: ProductDto;
  count: number;
}
